terraform/main.tf

```diff
@@ -0,0 +1,175 @@
+# operations like pulling Docker images and pushing logs to CloudWatch
+data "aws_iam_role" "ecs_task_execution_role" {
+  name = "ecsTaskExecutionRole"
+}
+
+# Retrieve the IAM role that the Batch job passes on to the containers, allowing
+# those containers to access AWS resources like data stored in S3
+data "aws_iam_role" "ecs_job_role" {
+  name = "ccao-ecs-model-runner"
+}
+
+# Create a Batch compute environment to run containers. Compute environments
+# define the underlying ECS or EC2 resources that will be provisioned to
+# use for running jobs in containers. Docs here:
+# https://docs.aws.amazon.com/batch/latest/userguide/compute_environments.html
+resource "aws_batch_compute_environment" "main" {
+  compute_environment_name = var.batch_job_name
+  service_role             = data.aws_iam_role.batch_service_role.arn
+  state                    = "ENABLED"
+  type                     = "MANAGED"
+
+  compute_resources {
+    type               = "FARGATE"
+    min_vcpus          = 0
+    max_vcpus          = 64  # Max across all jobs, not within one job
+    security_group_ids = [data.aws_security_group.outbound_https.id]
+    subnets            = data.aws_subnets.default.ids
+  }
+}
+
+# Create a Batch job queue to run jobs. Job queues keep track of which jobs
+# are waiting to run and in what order they should be prioritized in cases
+# where its associated compute environment has reached max capacity. Docs here:
+# https://docs.aws.amazon.com/batch/latest/userguide/job_queues.html
+resource "aws_batch_job_queue" "main" {
+  name                 = var.batch_job_name
+  compute_environments = [aws_batch_compute_environment.main.arn]
+  priority             = 0
+  state                = "ENABLED"
+}
+
+# Create a Batch job definition to define jobs. Job definitions provide the
+# information about how containers should be configured in a compute
+# environment and what those containers should do. When combined with a
+# job queue, they provide enough information to submit a job. Docs here:
+# https://docs.aws.amazon.com/batch/latest/userguide/job_definitions.html
+#
+# Note that jobs using this job definition cannot be provisioned by Terraform,
+# and are instead submitted via calls to `aws batch submit-job` in the same CI
+# workflow that provisions these resources
+resource "aws_batch_job_definition" "main" {
+  name                  = var.batch_job_name
+  platform_capabilities = ["FARGATE"]
+  type                  = "container"
+
+  container_properties = jsonencode({
+    executionRoleArn = data.aws_iam_role.ecs_task_execution_role.arn
+    fargatePlatformConfiguration = {
+      platformVersion = "LATEST"
+    }
+    image = var.batch_container_image_name
+    jobRoleArn = data.aws_iam_role.ecs_job_role.arn
+    logConfiguration = {
+      logDriver = "awslogs"
+    }
+    networkConfiguration = {
+      assignPublicIp = "ENABLED"
+    }
+    resourceRequirements = [
+      {
+        type = "VCPU"
+        value = "16.0"
```

Comment:
**question:** So we could presumably adjust the CPU count within a PR, as long as it's within the resource constraints of the compute environment?

Reply:
Yup, that's right!